### PR TITLE
Bump all versions to v2beta1 and re-use types from proto-public

### PR DESCRIPTION
**Pull request by ishustava**
Changes proposed in this PR:
- Once https://github.com/hashicorp/consul/pull/18930 gets merged, we need to update k8s to use latest versions
- With https://github.com/hashicorp/consul/pull/18935, we can re-use type definitions from the `proto-public` module

How I've tested this PR:

Ran unit tests locally with the consul binary built from ☝️ PRs. Note the pipeline tests will fail because those are not merged yet.

How I expect reviewers to test this PR:
👀 

Checklist:
- [ ] Tests added
- [ ] [CHANGELOG entry added](https://github.com/hashicorp/consul-k8s/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: control-plane/connect-inject/controllers/endpointsv2/endpoints_controller_test.go
Comment (zalimeni) [inline control-plane/connect-inject/controllers/endpointsv2/endpoints_controller_test.go:115]: 👍🏻

**Review by zalimeni — APPROVED** [review]
🙌🏻  these type references are great to have! LGTM +/- one question on the red diff

**Review by DanStough — APPROVED** [review]
Much cleaner! Would love to know the answer to @zalimeni 's question. My guess is we are finally default the `PeerName` in Consul?

**Comment by zalimeni** [comment]
Kicked CI now that https://github.com/hashicorp/consul/pull/18930 is merged. @ishustava any concern merging this to unblock open PRs?

**Comment by ishustava** [comment]
@zalimeni let me update proto-public to the latest from that PR and rebase. I'll merge then

File: control-plane/connect-inject/controllers/serviceaccount/serviceaccount_controller_test.go
Comment (ishustava) [reply control-plane/connect-inject/controllers/serviceaccount/serviceaccount_controller_test.go:146]: Not intentionally, but the test does pass for me locally 🤔

Comment (zalimeni) [reply control-plane/connect-inject/controllers/serviceaccount/serviceaccount_controller_test.go:146]: I think Dan's right that it's from the smoothing changes R.B. and Semir have made - either way, feel free to cut out of this one, we can get them all in a follow-up.

Comment (ishustava) [reply control-plane/connect-inject/controllers/serviceaccount/serviceaccount_controller_test.go:146]: I think it passes because we only compare the data part of the workload identity instead of the whole resource. I tried to fix it but it fails with peer name from tenancy missing from tenancy so I think it needs to be addressed in a separate PR.

Comment (zalimeni) [reply control-plane/connect-inject/controllers/serviceaccount/serviceaccount_controller_test.go]: Ah, thanks for the lead - and trying to fix. That makes sense 👍🏻

**Comment by ishustava** [comment]
Going to close this PR as we'd need to bump proto-public module before the release anyway.

**Pull request closed without merging**
